#### This notebook analyses the viscosity parameter for WCVI NEMO model. 

The runs with constant values of eddy laplacian viscosity are analysed at first and then the model is tested against the Smagorinksi viscosity which varies in our domain based on the relative sizes of the grid cells. 


#### Smagorinski is used only for lateral diffusion  of momentum (smag_coeff = 4)  whereas for tracers a constant laplacian diffusivity of 1000 is chosen and an eddy induced velocity coeffecient of 50

In [1]:
import matplotlib.pyplot as plt
import numpy as np
import numpy.ma as ma
import netCDF4 as nc
import cmocean.cm as cm 
from salishsea_tools import (
        viz_tools,
        nc_tools,
)
from IPython.display import display, Math, Latex
%matplotlib inline

/home/ssahu/anaconda3/lib/python3.5/site-packages/xarray/conventions.py:9: FutureWarning: The pandas.tslib module is deprecated and will be removed in a future version.
  from pandas.tslib import OutOfBoundsDatetime


### Question: When we are using a constant eddy viscosity where do the veloctites go to high values and what numbers do we get for the max velocity? 

To this extent we will analyse the 3 reduced viscosity runs (reduced from 1000)

In [3]:


visc_500_T = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_500_30_day_run/WCVI_1h_20160402_20160509_grid_T.nc')
visc_500_U = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_500_30_day_run/WCVI_1h_20160402_20160509_grid_U.nc')
visc_500_V = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_500_30_day_run/WCVI_1h_20160402_20160509_grid_V.nc')

visc_225_T = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_225_30_day_run/WCVI_1h_20160402_20160509_grid_T.nc')
visc_225_U = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_225_30_day_run/WCVI_1h_20160402_20160509_grid_U.nc')
visc_225_V = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_225_30_day_run/WCVI_1h_20160402_20160509_grid_V.nc')

visc_125_crash_T = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_125/WCVI_1h_20160402_20160405_grid_T.nc')
visc_125_crash_U = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_125/WCVI_1h_20160402_20160405_grid_U.nc')
visc_125_crash_V = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_125/WCVI_1h_20160402_20160405_grid_V.nc')


In [18]:
U_vel_500 = visc_500_U.variables['vozocrtx'][:];
V_vel_500 = visc_500_V.variables['vomecrty'][:];


u_unstgg_500, v_unstagg_500 = viz_tools.unstagger(U_vel_500, V_vel_500)
speed_500 = np.sqrt(np.square(u_unstgg_500) + np.square(u_unstgg_500))

print(np.max(speed_500),np.min(speed_500))

(t,z,y,x) = np.where(speed_500 == np.max(speed_500))

print(t,z,y,x)

1.33322 0.0
[257] [0] [97] [35]


### For the 500 run the max velocity (1.33 m/sec) were at the surface at (y,x) = (97,35)

In [19]:
U_vel_225 = visc_225_U.variables['vozocrtx'][:];
V_vel_225 = visc_225_V.variables['vomecrty'][:];


u_unstgg_225, v_unstagg_225 = viz_tools.unstagger(U_vel_225, V_vel_225)
speed_225 = np.sqrt(np.square(u_unstgg_225) + np.square(u_unstgg_225))

print(np.max(speed_225),np.min(speed_225))

(t,z,y,x) = np.where(speed_225 == np.max(speed_225))

print(t,z,y,x)

6.05275 0.0
[39] [0] [97] [37]


### For the 225 run the max velocity (6.05 m/sec) were at the surface at (y,x) = (97,37)

In [21]:
U_vel_125 = visc_125_crash_U.variables['vozocrtx'][:];
V_vel_125 = visc_125_crash_V.variables['vomecrty'][:];


u_unstgg_125, v_unstagg_125 = viz_tools.unstagger(U_vel_125, V_vel_125)
speed_125 = np.sqrt(np.square(u_unstgg_125) + np.square(u_unstgg_125))

print(np.max(speed_125),np.min(speed_125))

(t,z,y,x) = np.where(speed_125 == np.max(speed_125))

print(t,z,y,x)

13.9753 0.0
[59] [18] [82] [1]


### For the run crash of viscosity 125, the max velocities go to 13.9753 at the depth level 18 (out of 27) that too at the western boundary (to the top)

In [2]:
visc_smag_T = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_eiv_50_smag_4_30_day_run/WCVI_1h_20160402_20160509_grid_T.nc')
visc_smag_U = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_eiv_50_smag_4_30_day_run/WCVI_1h_20160402_20160509_grid_U.nc')
visc_smag_V = nc.Dataset('/ocean/ssahu/CANYONS/Results/NEMO_trial_viscosity/viscosity_eiv_50_smag_4_30_day_run/WCVI_1h_20160402_20160509_grid_V.nc')

In [3]:
U_vel_smag = visc_smag_U.variables['vozocrtx'][:];
V_vel_smag = visc_smag_V.variables['vomecrty'][:];


u_unstgg_smag, v_unstagg_smag = viz_tools.unstagger(U_vel_smag, V_vel_smag)
speed_smag = np.sqrt(np.square(u_unstgg_smag) + np.square(u_unstgg_smag))

print(np.max(speed_smag),np.min(speed_smag))

(t,z,y,x) = np.where(speed_smag == np.max(speed_smag))

print(t,z,y,x)

0.5515 0.0
[138] [0] [39] [58]


#### The run with smagorinksi viscosity has max horizontal speeds (0.55 m/sec) and it occurs at the surface 